contentreview-impl/tool/src/java/org/sakaiproject/contentreview/servlet/GradingCallbackServlet.java

```diff
@@ -0,0 +1,165 @@
+import javax.servlet.http.HttpServletResponse;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import org.sakaiproject.assignment.api.Assignment;
+import org.sakaiproject.assignment.api.AssignmentContent;
+import org.sakaiproject.assignment.api.AssignmentSubmissionEdit;
+import org.sakaiproject.assignment.cover.AssignmentService;
+import org.sakaiproject.component.cover.ComponentManager;
+import org.sakaiproject.contentreview.model.ContentReviewItem;
+import org.sakaiproject.contentreview.service.ContentReviewService;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.cover.SessionManager;
+
+/** 
+ * This servlet will receive callbacks from TII. Then it will process the data
+ * related to the grades and store it.
+ */
+
+@SuppressWarnings("deprecation")
+public class GradingCallbackServlet extends HttpServlet {
+	
+	private static Log M_log = LogFactory.getLog(GradingCallbackServlet.class);
+	
+	private ContentReviewService contentReviewService =
+				(ContentReviewService)ComponentManager.get("org.sakaiproject.contentreview.service.ContentReviewService");
+	
+	@Override
+	public void init(ServletConfig config) throws ServletException {
+		M_log.debug("init GradingCallbackServlet");
+		super.init(config);
+	}
+	
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		M_log.debug("doGet GradingCallbackServlet");
+		doPost(request, response);
+	}
+	
+	@SuppressWarnings("unchecked")
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		M_log.debug("doPost GradingCallbackServlet");
+		String contentType = request.getContentType();
+		if ( contentType != null && contentType.startsWith("application/xml") ) {
+			doPostXml(request, response);
+		} else {
+			M_log.warn("GradingCallbackServlet received a not json call.");
+			response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+		}
+	}
+	
+	@SuppressWarnings("unchecked")
+    protected void doPostXml(HttpServletRequest request, HttpServletResponse response) 
+            throws ServletException, IOException 
+    {
+		M_log.debug("doPostXml GradingCallbackServlet");
+		StringBuilder sb = new StringBuilder();
+		BufferedReader reader = request.getReader();
+		try {
+			String line;
+			while ((line = reader.readLine()) != null) {
+				sb.append(line).append('\n');
+			}
+		} finally {
+			reader.close();
+		}
+		M_log.debug(sb.toString());
+		
+		DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+		String sourcedId = null;
+        try{
```

Comment:
I don't see any authentication on the callback here, what's to spot someone making fake requests against this endpoint?